tuf/api/metadata.py

```diff
@@ -1384,8 +1490,39 @@ def _is_target_in_pathpattern(targetpath: str, pathpattern: str) -> bool:
 
         return True
 
+    def find_delegation(self, target_filepath: str) -> Optional[str]:
+        """Find the delegated rolename based on the given ``target filepath``.
+
+        If the given ``target_filepath`` is not in any of the paths that
+        ``DelegatedRole`` is trusted to provide it returns None.
+
+        Args:
+            target_filepath: URL path to a target file, relative to a base
+                targets URL.
+        """
+        # Check if the given target_filepath is in any of the "paths" or
+        # "path_hash_prefixes" that DelegatedRole is trusted to provide.
+        if self.is_delegated_path(target_filepath):
+            return self.name
+
+        if self.succinct_hash_info is not None:
+            # Split hash calculation and from the rest for easier testing.
+            hasher = sslib_hash.digest(algorithm="sha256")
+            hasher.update(target_filepath.encode("utf-8"))
+            # Get the bit representation of the hash.
+            hash_bits = "".join(
+                f"{one_byte:08b}" for one_byte in hasher.digest()
+            )
+
+            # pylint: disable=protected-access
+            return self.succinct_hash_info._calculate_delegation(hash_bits)
+
+        return None
+
     def is_delegated_path(self, target_filepath: str) -> bool:
-        """Determines whether the given ``target_filepath`` is in one of
+        """This method is depricated, use find_delegation instead!
```

Comment:
```suggestion
        """This method is deprecated, use ``find_delegation()`` instead.
```

fixed the typo, but while I was doing that also removed the exclamation: no need to shout in docs. Same typo in a warning message.

There might be some fancy way to product nice note/warning layout in RTD... but I think this is fine, no need to overdo it